src/domain/select_local_cameras.py

```diff
@@ -183,6 +183,7 @@ def accept_and_close(self):
                 checkbox = self.findChild(QCheckBox, f"checkBox_camera_{idx}")
                 line_edit = self.findChild(QLineEdit, f"lineEdit_cameraID_{idx}")
                 if checkbox and line_edit:
+                    saved = False
                     for camera in cameras:
                         if camera.type == Camera.CameraTypes.USBCamera:
```

Comment:
TODO: define a method to avoid type checks on (sub)classes.

Reply:
for example?

Reply:
Looking at the overall code, in this case are needed more methods.
- for weak equality: self.enumerated_usb_cameras[idx].name == camera.name
                                and self.enumerated_usb_cameras[idx].pid == camera.pid
                                and self.enumerated_usb_cameras[idx].vid == camera.vid
                                and self.enumerated_usb_cameras[idx].path == camera.path
- for stronger equality (adding the index);
- for copying some attributes from one instance to another one.

Reply:
Yes, usb camera enumeration is complex as same camera connected to different bus/ports on the pc might be re-indexed. We could sync quickly on weekend so that I can show you all corner cases I had to cover. Improvements are of course welcome.